### PR TITLE
Check private methods are not installed before super returns

### PR DESCRIPTION
**Pull request by mkubilayk**
Adds tests calling private methods before super is constructed - according to the test plan posted in #1343.

>  Private methods are installed "when super returns" and no earlier. A similar test is possible with field initializers instead of code called in the constructor.

I will commit the generated tests once the PR is approved.

@leobalter @rwaldron for review.
cc: @littledan @jbhoosreddy @caiolima

### CODE REVIEW

**Comment by mkubilayk** [comment]
@caiolima updated with tests covering private accessors. I thought we generate tests once the PR is approved. I sent them as a separate commit just in case.

File: src/class-elements/prod-private-getter-before-super-return-in-constructor.case [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-getter-before-super-return-in-constructor.case]: nit: This trailing line makes YAML break sometimes. Could you remove them?

File: src/class-elements/prod-private-getter-before-super-return-in-field-initializer.case [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-getter-before-super-return-in-field-initializer.case]: ditto.

File: src/class-elements/prod-private-method-before-super-return-in-field-initializer.case [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-method-before-super-return-in-field-initializer.case]: Ditto.

File: src/class-elements/prod-private-setter-before-super-return-in-constructor.case [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-setter-before-super-return-in-constructor.case]: ditto.

File: src/class-elements/prod-private-method-before-super-return-in-constructor.case [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-method-before-super-return-in-constructor.case]: ditto.

File: src/class-elements/prod-private-setter-before-super-return-in-field-initializer.case [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    8. Let F be thisER.[[FunctionObject]].
+    9. Assert: F is an ECMAScript function object.
+    10. Perform ? InitializeInstanceElements(result, F).
+
+  EDITOR'S NOTE:
+    Private fields are added to the object one by one, interspersed with
+    evaluation of the initializers, following the construction of the
+    receiver. These semantics allow for a later initializer to refer to
+    a previous private field.
+
```
Comment (caiolima) [inline src/class-elements/prod-private-setter-before-super-return-in-field-initializer.case]: ditto.

**Comment by caiolima** [comment]
> I thought we generate tests once the PR is approved. I sent them as a separate commit just in case.

Oh right. I don't think there is a strict rule here, but IMHO it is good to have them generated on first PR, because some templates are hard to review without generated tests. Also, the CI checks if they are missing.

**Comment by mkubilayk** [comment]
@caiolima thanks for the review. Removed the trailing blank lines.

File: src/class-elements/prod-private-getter-before-super-return-in-constructor.case [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+---*/
+
+//- elements
+constructor() {
+    this.f();
+}
+
+//- assertions
+class D extends C {
+    f() { this.#m; }
```
Comment (caiolima) [inline src/class-elements/prod-private-getter-before-super-return-in-constructor.case]: nit: Oops, I missed this in the first review, but I think the code style uses 2 space bars.

Comment (leobalter) [reply src/class-elements/prod-private-getter-before-super-return-in-constructor.case]: well, this doesn't change the test and there is more than just this to get fixed. 

I'd say we defer style for a follow up or a general fix in the test files.

**Comment by mkubilayk** [comment]
Thanks!

**Pull request merged**